python/gsw/main.py

```diff
@@ -0,0 +1,303 @@
+from flask import Flask, request, jsonify
+from flasgger import Swagger
+from flasgger.utils import swag_from
+from elasticsearch import Elasticsearch
+from datetime import datetime
+import threading
+import json
+import imaplib
+import base64
+import os
+import email
+import logging
+
+app = Flask(__name__)
+app.config["SWAGGER"]={"title": "PAN Ground Software", "uiversion": 2}
+ 
+class read_iridium(object):
+    def __init__(self, radio_keys_config, server_keys_config):
+        #elasticsearch server
+        self.es_server=server_keys_config["server"]
+        self.es_port=server_keys_config["port"]
+
+        #pan email
+        self.username=radio_keys_config["email_username"]
+        self.password=radio_keys_config["email_password"]
+
+        #updates MOMSN and MTMSN numbers sent/recieved
+        self.momsn=-1
+        self.mtmsn=-1
+        self.confirmation_mtmsn=-1
+
+        #send_uplinks, keeps track of if the ground can send more uplinks. This allows us to make sure we are only sending one uplink at a time.
+        self.send_uplinks=False
+
+        #connect to email
+        self.mail = imaplib.IMAP4_SSL("imap.gmail.com", 993)
+        self.mail.login(self.username, self.password)
+        self.mail.select('Inbox')
+
+        #thread
+        self.run_email_thread = False
+
+    def connect(self):
+        '''
+        Starts a thread which will continuously
+        check the PAN email and post reports to 
+        elasticsearch 
+        '''
+        self.check_email_thread = threading.Thread(target=self.post_to_es)
+        self.run_email_thread = True
+        self.check_email_thread.start()
+
+    def process_downlink_packet(self, data):
+        '''
+        Converts the email attachment data into a 
+        JSON object. This is because elasticsearch
+        only takes in JSON data. I will add to this
+        once downlink producer is finished. 
+        '''
+        return json.loads(data)
+
+    def check_for_email(self):
+        '''
+        Checks the PAN email account. Updates the most recent MOMSN 
+        and MTMSN numbers. Returns the contents of the email attachement
+        of the most recent unread email from the Iridium email account.
+        '''
+        #look for all new emails from iridium
+        _, data = self.mail.search(None, '(FROM "fy56@cornell.edu")', '(UNSEEN)')
```

Comment:
Should this be changed to something more general eventually? (already talked to Fatima about this)